cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +34,67 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_main_package_name_and_version(package_dir, dependencies)
     return []
+
+
+def _get_main_package_name_and_version(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """
+    Get main package name and version.
+
+    The main package is the package that is being processed by cachi2.
+    Not any of its dependencies.
+    """
+    path_dependencies = [dep for dep in dependencies if isinstance(dep, PathDependency)]
+    name_and_version = _get_name_and_version_from_path_deps(path_dependencies)
+    if name_and_version is not None:
+        return name_and_version
+
+    log.info("Failed to infer name and version from gemspec file")
+
+    # fallback to origin remote
+    try:
+        name = _get_repo_name_from_origin_remote(package_dir)
+    # if the git repository does not have an origin remote
+    except UnsupportedFeature:
+        raise PackageRejected(
+            reason="Failed to infer package name from origin remote",
+            solution=(
+                "Please specify package name and version in a way that Cachi2 understands,\n"
+                "or make sure that the directory Cachi2 is processing is a git repository with\n"
+                "an 'origin' remote, in which case Cachi2 will infer the package name from the remote URL."
+            ),
+        )
+
+    return name, None
+
+
+def _get_name_and_version_from_path_deps(
+    path_dependencies: list[PathDependency],
+) -> Optional[tuple[str, str]]:
+    """
+    Extract name and version from path dependencies in the Gemfile.lock.
+
+    If the gemspec file exists in the package directory and is explicitly defined in the Gemfile,
+    then the Gemfile.lock will contain a path dependency representing the gem in the package directory.
```

Comment:
This is a bit incorrect. First, there is no reference to back this statement and I had to go digging what this is based on. What I've come across was this https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/ (not an official doc, but gave me some needed perspective). So the logic you have here assumes that `Gemfile.lock` is going to be present at all times and if it happens that `.gemspec` will be present as well, we'll use the info - which based on the article will be like lower units of % of cases because it is not recommended to check in one's `Gemfile.lock` when developing gems while it's equally unusual for apps to specify a gemspec when they're technically not a gem.

So, I'd expect the docstring to be improved a bit such that it mentions that what it's doing would happen in a fairly unlikely scenario. Additionally, mention our design doc as a basis for the implementation (we do have a note on gemspec path dependency)

Reply:
I never seen a repo without `Gemfile.lock`. Isn't the purpose of that file to ensure consistency and that everyone has the same dependencies installed? Similarly, as it is, for example, for pipenv `Pipfile.lock`.
https://pipenv-fork.readthedocs.io/en/latest/basics.html#pipfile-lock-security-features

Reply:
For example: https://github.com/ruby-git/ruby-git

It's explained in the article I linked. We would face issues if people tried to build gems with cachi2. Then again, in that case `bundler` would most likely not be involved. That said, the fact remains that having both files is a very rare scenario.

Reply:
Okay, so simultaneously having `gemspec` and `Gemfile.lock` is unusual. Got it.
Now it makes sense. Docs updated.